craft_parts/infos.py

```diff
@@ -0,0 +1,211 @@
+        plugin_version: str = "v2",
+        project_dirs: ProjectDirs = None,
+        **custom_args,  # custom passthrough args
+    ):
+        if not project_dirs:
+            project_dirs = ProjectDirs()
+
+        self._application_name = application_name
+        self._set_machine(arch)
+        self._plugin_version = plugin_version
+        self._parallel_build_count = parallel_build_count
+        self._dirs = project_dirs
+        self._custom_args = custom_args
+
+    def __getattr__(self, name):
+        if hasattr(self._dirs, name):
+            return getattr(self._dirs, name)
+
+        if name in self._custom_args:
+            return self._custom_args[name]
+
+        raise AttributeError(f"{self.__class__.__name__!r} has no attribute {name!r}")
+
+    @property
+    def custom_args(self) -> List[str]:
+        """Return the list of custom argument names."""
+        return list(self._custom_args.keys())
+
+    @property
+    def application_name(self) -> str:
+        """Return the name of the application using craft-parts."""
+        return self._application_name
+
+    @property
+    def arch_triplet(self) -> str:
+        """Return the machine-vendor-os platform triplet definition."""
+        return self.__machine_info["triplet"]
+
+    @property
+    def is_cross_compiling(self) -> bool:
+        """Whether the target and host architectures are different."""
+        return self.__target_machine != self.__platform_arch
+
+    @property
+    def plugin_version(self) -> str:
+        """Return the plugin API version used in this project."""
+        return self._plugin_version
+
+    @property
+    def parallel_build_count(self) -> int:
+        """Return the maximum allowable number of concurrent build jobs."""
+        return self._parallel_build_count
+
+    @property
+    def target_arch(self) -> str:
+        """Return the architecture used for deb packages."""
+        return self.__machine_info["deb"]
+
+    @property
+    def dirs(self) -> ProjectDirs:
+        """Return the project's work directories."""
+        return self._dirs
+
+    @property
+    def project_options(self) -> Dict[str, Any]:
+        """Obtain a project-wide options dictionary."""
+        return {
+            "application_name": self.application_name,
+            "arch_triplet": self.arch_triplet,
+            "target_arch": self.target_arch,
+        }
+
+    def _set_machine(self, target_arch):
+        self.__platform_arch = _get_platform_architecture()
+        if not target_arch:
+            target_arch = self.__platform_arch
+            logger.debug("Setting target machine to %s", target_arch)
+
+        machine = _ARCH_TRANSLATIONS.get(target_arch, None)
+        if not machine:
+            raise errors.InvalidArchitecture(target_arch)
+
+        self.__target_machine = target_arch
+        self.__machine_info = machine
+
+
+def _get_platform_architecture() -> str:
+    # TODO: handle Windows architectures
```

Comment:
AFAIK `platform.machine()` also works in Windows.

Reply:
We'll probably need extra handling here to tell that this is Windows along with the architecture. I think this will become something more like `_get_host_platform()` instead of a plain `_get_host_architecture()`.

Reply:
ok